js/sparkswap.js

```diff
@@ -173,7 +173,50 @@ module.exports = class sparkswap extends Exchange {
     }
 
     async fetchMarkets () {
-        throw new ExchangeError ('Not Implemented');
+        let response = await this.privateGetV1Markets ();
+        let markets = response['supported_markets'];
+        let result = [];
+        for (let i = 0; i < markets.length; i++) {
+            const market = markets[i];
+            const id = market['id'];
+            const symbolParts = id.split ('/');
+            const baseId = symbolParts[0];
+            const quoteId = symbolParts[1];
+            const base = this.commonCurrencyCode (baseId.toUpperCase ());
```

Comment:
2 thoughts on this. the code `this.commonCurrencyCode` works because of a pass through, however we are not actually using this method correctly.

We can either 1. remove this.commonCurrencyCode or 2. define `commonCurrencies` in `sparkswap.js`. An example of usage is here: https://github.com/ccxt/ccxt/blob/a0eb173de40c09cf06d0624c1f677ef22ffcfdfa/js/ccex.js#L72-L79.